lib/griddler/mailgun/adapter.rb

```diff
@@ -80,12 +82,23 @@ def param_or_header(key)
       end
 
       def attachment_files
-        attachment_count = params['attachment-count'].to_i
-
-        attachment_count.times.map do |index|
-          params.delete("attachment-#{index+1}")
+        attachments = JSON.parse(params['attachments']) || Array.new
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.